tests/console/validate_selected_patterns.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validation module to check patterns installed.
+# Scenarios covered:
+# - zypper search for named patterns(base,enhanced_base;
+# - zypper search for removed pattern apparmor
+#
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+
+sub run {
+
+    select_console('root-console');
+
+    my @installed_patterns = split /,/, get_var('PATTERNS');
+    my %patterns           = (
+        base     => 'base',
+        minimal  => 'enhanced_base',
+        apparmor => 'apparmor',
+        x        => 'x11',
+        yast2    => 'yast2_basis'
+    );
+    foreach my $key (keys %patterns) {
+        if (grep /^$key$/, @installed_patterns) {
+            record_info("Verify installed", "$patterns{$key}");
+            zypper_call("se -x -t pattern -i -n $patterns{$key}");
+        }
+        else {
+            record_info("Verify not installed", "$patterns{$key}");
```

Comment:
I've checked `tests/installation/select_patterns_and_packages.pm` and now I remember that `PATTERNS` can have `-` to state that we won't to deselect pattern, so we will get an error as will try to concatenate undef.
For the cases where we don't have such key I would suggest just report that pattern is not tested, and validate those with minus key, that they are not installed.
Also, I guess it makes more sense to go through the list of `@installed_patterns` and not keys, as then we can reduce complexity to O(n) instead of O(n^2) with grep.

Reply:
The purpose of %patterns is to satisfy "and possibly (as negative test), that unselected patterns are not installed." that's the reason to loop through the %patterns and not the @installed_patterns. How do you think i can do the negative test if i dont go through a check. i can only think explicit zypper call as it was in the first place.

As for `-`, i havent see use of it. So if it appears and the package is selected, this pattern should remain selected and if it not already by default then it remains as it is(?). I cant understand what do you mean with "For the cases where we don't have such key I would suggest just report that pattern is not tested, and validate those with minus key, that they are not installed."

Reply:
So i have modified the regex to consider the minus in front of the patterns and if it doesnt match it will go through the else statement.